### PR TITLE
langchain[minor]: add EnsembleRetriever

### LINKED ISSUE #5554: DOC:  How to merge or use multiple retrievers?
### Checklist

- [X] I added a very descriptive title to this issue.
- [X] I included a link to the documentation page I am referring to (if applicable).

### Issue with current documentation:

_No response_

### Idea or request for content:

The documentation misses this use case. Based of search seems python has the merger_retriever but js doesn't

### PR DESCRIPTION
**Pull request by jeasonnow**
Fixes #5554

### CODE REVIEW

**Comment by jacoblee93** [comment]
Thank you! This needs some docs, but I can add some myself

File: langchain/src/retrievers/multi_retriever.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import { BaseDocumentCompressor } from "@langchain/core/retrievers/document_compressors";
+
+export interface MultiRetrieverInput extends BaseRetrieverInput {
+  retrivers: BaseRetriever[];
+  documentCompressor?: BaseDocumentCompressor | undefined;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  documentCompressorFilteringFn?: (docs: Document<Record<string, any>>[]) => Document<Record<string, any>>[];
+}
+
+export class MultiRetriever extends BaseRetriever {
```
Comment (jacoblee93) [inline langchain/src/retrievers/multi_retriever.ts:13]: Can we rename to `EnsembleRetriever` to match Python?

https://python.langchain.com/v0.2/docs/how_to/ensemble_retriever/

File: langchain/src/retrievers/multi_retriever.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    this.retrievers = args.retrivers;
+    this.documentCompressor = args.documentCompressor;
+    this.documentCompressorFilteringFn = args.documentCompressorFilteringFn;
+  }
+
+  async _getRelevantDocuments(
+    query: string,
+    runManager?: CallbackManagerForRetrieverRun
+  ) {
+    return this._merge_documents(query, runManager);
```
Comment (jacoblee93) [inline langchain/src/retrievers/multi_retriever.ts:38]: nit: prefer camel case

File: langchain/src/retrievers/multi_retriever.ts (hunk 1 above)
Comment (jacoblee93) [inline langchain/src/retrievers/multi_retriever.ts:13]: Might be worthwhile to implement the `EnsembleRetriever` instead?

https://python.langchain.com/v0.2/docs/how_to/ensemble_retriever/

Comment (jeasonnow) [reply langchain/src/retrievers/multi_retriever.ts:13]: That looks good, I will implement it later.

File: langchain/src/retrievers/multi_retriever.ts (hunk 2 above)
Comment (jeasonnow) [reply langchain/src/retrievers/multi_retriever.ts]: done

**Comment by jeasonnow** [comment]
I've created an `EnsembleRetriever` that's functionally identical to the Python version of `ensemble_retriever`. Would you mind taking another look at it? I'll fill in the documentation when I have some time.  @jacoblee93

**Pull request merged**